### PR TITLE
Erp federation

### PR DESCRIPTION
**Pull request by marcos-iov**
*fed:erp-fed-integration*
*fit:adapt-test-for-erp*

### CODE REVIEW

**Comment by donequis** [comment]
Something changed on the generation of the verification data. It now ignores the pom files

**Comment by marcos-iov** [comment]
> Something changed on the generation of the verification data. It now ignores the pom files

We were having some issues recreating the verifications file, it's solved now. Thanks for the heads up!

File: rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java [hunk 1]
```diff
@@ -262,8 +265,18 @@ public static int countMissingSignatures(Context btcContext, BtcTransaction btcT
         TransactionInput input = btcTx.getInput(0);
         Script scriptSig = input.getScriptSig();
         List<ScriptChunk> chunks = scriptSig.getChunks();
+        Script redeemScript = new Script(chunks.get(chunks.size() - 1).data);
```
Comment (donequis) [inline rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java:268]: This method seems to be used only for logging. We should make sure it is only called if logging level is enabled.

Comment (donequis) [reply rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java:268]: This method = `countMissingSignatures`

**Comment by marcos-iov** [comment]
pipeline:run

**Comment by marcos-iov** [comment]
pipeline:run

**Comment by guidohernan93** [comment]
pipeline:run

**Comment by guidohernan93** [comment]
pipeline:run

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java [hunk 2]
```diff
@@ -441,6 +448,20 @@ protected TxType getTransactionType(BtcTransaction btcTx) {
         return TxType.UNKNOWN;
     }
 
+    private boolean txIsFromOldFederation(BtcTransaction btcTx) {
+        Address oldFederationAddress = Address.fromBase58(bridgeConstants.getBtcParams(), bridgeConstants.getOldFederationAddress());
```
Comment (donequis) [inline rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java:452]: Shouldn't this check for an oldFederation in storage and only check the bridge constant if it does not exist?

File: rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java (hunk 1 above)
Comment (marcos-iov) [reply rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java:268]: Done

**Comment by marcos-iov** [comment]
pipeline:run

**Comment by marcos-iov** [comment]
pipeline:run

**Review by josedahlquist — APPROVED** [review]

**Pull request merged**